mongo/integration/client_test.go

```diff
@@ -417,7 +418,12 @@ func TestClient(t *testing.T) {
 		// First two messages should be connection handshakes: one for the heartbeat connection and the other for the
 		// application connection.
 		for idx, pair := range msgPairs[:2] {
-			assert.Equal(mt, pair.CommandName, "isMaster", "expected command name isMaster at index %d, got %s", idx,
+			helloCommand := "isMaster"
+			//  Expect "hello" command name with API version.
+			if os.Getenv("REQUIRE_API_VERSION") == "true" {
+				helloCommand = "hello"
+			}
```

Comment:
I'm concerned that determining whether or not to enforce the versioned API here and where the `Client` is configured could get out-of-sync and cause this test to succeed when it should fail. E.g. if the client is configured with versioned API enabled but this test still asserts that command `"isMaster"` is sent when we're actually expecting `"hello"`.

Is there a more direct way to determine if the `Client` was configured with the versioned API enabled that won't get out-of-sync with the `Client` configuring code in `mtest.T#createTestClient`?

Reply:
Hmm I'm actually not sure of a better way. `serverAPI` is a private field on `Client`, so it cannot be checked to use in this conditional. `testContext` holds the `requireAPIVersion` bool that decides whether to add server API options to the test client, but that is also a private variable in the `mtest` package. I think your concern is valid, but this conditional will be removed once `isMaster` is no longer present in the codebase (GODRIVER-1862).